editquality/utilities/generate_make.py

```diff
@@ -31,34 +37,38 @@
 # * Where can we store information about samples?
 #   Original population rates; how we've distorted them.
 
+import logging
+import sys
+
 import docopt
-import os.path
 
-from ..codegen import differ
-from ..codegen import generate
+from ..codegen import differ, generate
 
-DEFAULT_BASE_DIR = os.path.dirname(__file__) + "/../.."
+logger = logging.getLogger(__name__)
 
 
 def main(argv=None):
     args = docopt.docopt(__doc__, argv=argv)
 
-    config_dir = args["--config-dir"] \
-        if args["--config-dir"] is not None else DEFAULT_BASE_DIR + "/config"
+    logging.basicConfig(
+        level=logging.DEBUG if args['--debug'] else logging.WARNING,
+        format='%(asctime)s %(levelname)s:%(name)s -- %(message)s'
+    )
+
+    config_path = args["--config"]
 
-    output_path = args["--output-path"] \
-        if args["--output-path"] is not None \
-        else DEFAULT_BASE_DIR + "/Makefile.automated"
+    output_f = sys.stdout \
+        if args["--output"] == "<stdout>" \
```

Comment:
IMO these inline ternaries are gross, but it looks like I started the trend here ;-). Coming back as a reviewer, it's confusing.